docs/Reference-for-template.json.md

```diff
@@ -52,31 +52,275 @@ You can find a sample of a complex `template.json` file here [Sample](https://gi
-||`Computed`: A symbol for which the config provides a Boolean predicate whose evaluation result is the computed symbol result||
-||`Generated`: A symbol whose value gets computed by a built-in symbol value generator. See [Details](https://github.com/dotnet/templating/wiki/Reference-for-available-macros)||
-|dataType |Indicates limitations on the valid values a symbol may be assigned. At this point, the only valid datatype is "choice", which also requires providing symbols.choices configuration for the symbol|`type=parameter`|
-|defaultValue|The value assigned to the symbol if no value for it is provided by the user or host|`type=parameter`|
-|binding|||
-|replaces|Text value that will be replaced||
-|fileRename|The fileRename element defines the filenames which will be replaced|`type=parameter`|
-|description|Human readable text describing the meaning of the symbol. This has no effect on template generation||
-|isRequired|Indicates if the parameter is required or not.|Only used when `type` is set to `parameter`|
-|choices|List of available choices.|Only used when `type` is `choice`|
-|value |Value for replacement|`type=computed`|
+There are the following symbol types:
+- Parameter
+- Derived
+- Computed
+- Generated
+- Bind
 
 There are 3 places from which a symbol can acquire its value:
+- Template configuration (this).
+- Host provided values, which override template configuration values. For example, the host may provide the operating system kind as "Linux", overriding a template default value of "Windows".
+- User provided values, which override host & template values. The way these are provided to the template generation broker is specific to the broker. This may be via additional configuration files, command line parameters, inputs to a UI, etc.
+
+#### Parameter symbol
+
+A symbol for which the config provides literal and/or default values.
+
+|Name|Description|
+|---|---|
+|`type`|`parameter`|
+|`dataType`|	Supported values: <br />- `bool`: boolean type, possible values: `true`/`false`. <br />- `choice`: enumeration, possible values are defined in `choices` property.<br />- `float`: double-precision floating format number. Accepts any value that can be parsed by `double.TryParse()`.<br />- `int`/`integer`: 64-bit signed integer. Accepts any value that can be parsed by `long.TryParse()`.<br />- `hex`: hex number. Accepts any value that can be parsed by `long.TryParse(value.Substring(2), NumberStyles.HexNumber, CultureInfo.InvariantCulture, out long convertedHex)`.<br />- `text`/`string`: string type.<br />- `<any other>`: treated as string.
+|`defaultValue`|The value assigned to the symbol if no value for it is provided by the user or host.|
+|`binding`|The name of the host property to take the value from.|	
+|`replaces`|The text to replace with the value of this symbol.|	 
+|`fileRename`|The fileRename element defines the filenames which will be replaced by symbol value.| 
```

Comment:
Is this the file name, or a portion of the filename. Is it "portions of filenames" rather than "filenames"?

Reply:
This is a portion of filename. For example if filename in template is `FooBar.cs`, and `fileRename=Foo`, the outcome will be `<value of symbol>Bar.cs`.